### DISCUSSION TITLE
Get pin locations

### DISCUSSION THREAD

**Discussion by young03600** [The-OpenROAD-Project/OpenLane#2174] [Q&A]
Hi,

I've searched some posted issues but not find the one.
I would like to know how can I extract all pin locations including components pins and PI/PO ports by given the DEF file?

Any help is appreciated. Thanks.

**Comment by kareefardi** [comment]
Hey @young03600. A DEF file has a `PINS` section that has all pin locations. What are you after exactly? That way we can assist you better.

**Reply by young03600** [reply]
Hi,
Thanks for your reply.

I've checked the ```PINS``` section you mentioned.
Sorry for the imprecise illustration. I've revised it.
I want ```all of the components pins``` not only the ```PI/PO ports```.

Thanks.
